sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -104,3 +110,37 @@ def test_md5sums():
         assert_false(sg.check_md5sum_change(f.name))
 
     os.remove(f.name + '.md5')
+
+
+def test_pattern_matching():
+    """Test if only examples matching pattern are executed"""
```

Comment:
My personal preference: I tend to put this kind of things as comment rather than docstrings. The reason: nosetests use the docstring rather than the name of the function in case of failure.

```
======================================================================
ERROR: Test if a known example file gets properly split
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/volatile/le243287/miniconda3/lib/python3.4/site-packages/nose/case.py", line 198, in runTest
```

In general the first thing I want to do in case of failure is to run `nosetests -m` to only run the failing test. `nosetests -m` is based on the name of the test function and I find it mildly annoying to do a `git grep -p' to figure out the name of the function from the docstring.